### DISCUSSION TITLE
How to generate a model with an enum?

### DISCUSSION THREAD

**Discussion by Kernald** [loco-rs/loco#854] [General]
I'm starting a toy project with Loco, and somehow all the resources I want to model contain enums. I've seen #782, but it's unclear to me where I should define these enums so they'd be picked up by the `loco` cli tool?

**Comment by jtwaleson** [comment]
In the migration, just select a string as the data type. Then run `cargo loco db entities` and in the `_entities/model.rs` file, add something like this:

```

// WARNING: do not overwrite this part of the file, the automatic
// Loco code gen can overwrite it
#[derive(Debug, Clone, PartialEq, Eq, EnumIter, DeriveActiveEnum, Serialize, Deserialize)]
#[sea_orm(rs_type = "String", db_type = "String(StringLen::N(32))")]
pub enum Role {
    #[sea_orm(string_value = "Admin")]
    Admin,
    #[sea_orm(string_value = "Editor")]
    Editor,
    #[sea_orm(string_value = "Viewer")]
    Viewer,
}

then in the type below change String to (in this case) Role.

pub struct Model {
    pub created_at: DateTimeWithTimeZone,
    pub updated_at: DateTimeWithTimeZone,
    // WARNING: do not overwrite this part of the file, the automatic
    // Loco code gen can overwrite it
    pub role: Role,
    ....
}
```

**Comment by liamrohan** [comment]
Would be really nice to run `loco generate scaffold` and have the CLI do this automatically: 

```shell
cargo loco generate scaffold customer \
  first_name:string! \
  # ...
  type:enum[Buyer,Seller] --api
```

Where `type:enum[Buyer,Seller]` would automatically generate the `CustomerType` `enum` inside `_entities/customer.rs`.

**Reply by kaplanelad** [reply]
This PR https://github.com/loco-rs/loco/pull/1568 has been merged and will be included in the next release.

**Reply by liamrohan** [reply]
Awesome. Thanks
